packages/jest-diff/README.md

```diff
@@ -0,0 +1,238 @@
+# jest-diff
+
+Display differences clearly so people can review changes confidently.
+
+The default export serializes JavaScript **values** and compares them line-by-line.
+
+Two named exports compare **strings** character-by-character:
+
+- `diffStringsAligned` returns a string which includes comparison lines.
```

Comment:
It took me quite a while (even after reading the whole readme) to understand the difference between those two. I still do not quite understand the naming, which I think was part of what confused me because I associated things like the beginning of lines with "aligned".
Would you say that unified diff vs split diff (like for example in GitHub) is a good analogy for those functions and could be used for naming?

Reply:
Agreed on being a bit confused about what `aligned` means. Unified vs split makes more sense to me (that's also the naming bitbucket uses, so I think it's pretty standard)

Reply:
Brilliant analogy!

* so-called aligned is like unified diff
* so-called unaligned is used in Jest to display one-line strings in normal compact report:

```text
Expected: "change from"
Received: "change to"
```

For multiline strings, a split like GitHub would consist of array of string tuples:

```js
[
  [a0, b0],
  [a1, b1],
  // and so on
]
```

for corresponding changed chunks. For deleted `b` is empty, for inserted `a` is empty.

I am curious, can y’all think of a use case in third-party dependents?

Reply:
* I have renamed `diffStringsAligned` and `diffStringsUnified`
* I think that I will replace `diffStringsUnaligned` with `diffStringsRaw` that returns an array of `Diff` objects, so that it will be reusable to the max